File: 체육복.py
def solution(n, lost, reserve):
    answer = 0
    visited=[1]*n
    for i in lost:
        visited[i-1]=0
        
    reserve.sort()
    for i in reserve:
        if i in lost:
            visited[i-1]=1
            reserve.remove(i)
    
    
    for i in reserve:
        if visited[i-1]==0:
            visited[i-1]=1
        else:
            if i==1 and visited[i]==0:
                visited[i]=1
            elif i==n and visited[i-2]==0:
                visited[i-2]=1    
            elif i!=1 and i!=n and visited[i-2]==0 and visited[i]==0:
                visited[i-2]=1
            elif i!=1 and i!=n and visited[i-2]==0 and visited[i]==1:
                visited[i-2]=1
            elif i!=1 and i!=n and visited[i-2]==1 and visited[i]==0:
                visited[i]=1
            elif i!=1 and i!=n and visited[i-2]==1 and visited[i]==1:
                continue
            
            else:
                continue
    
    
    return sum(visited)

File: boj_java/src/boj_java/체육복.java
import java.util.Arrays;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n;
        int[] student= new int[n];
        
        // 잃어버린 사람
        for(int i: lost){
            student[i-1]--;
        }
        
        // 여분 가진 사람
        for(int i: reserve){
            student[i-1]++;
        }
        
        // 체육복 빌리기
        for(int i=0;i<n;i++){
            if(student[i]<0){
                if(i!=0 && student[i-1]==1){
                    student[i-1]--;
                    student[i]++;
                }
                else if(i!=n-1 && student[i+1]==1){
                    student[i+1]--;
                    student[i]++;
                }
            }
        }
        for(int i:student){
            if(i==-1){
                answer--;
            }
        }
        
        return answer;
    }
}